tests/python_package_test/test_basic.py

```diff
@@ -332,6 +332,133 @@ def check_asserts(data):
     check_asserts(lgb_data)
 
 
+def test_category_encoding(tmp_path):
+
+    def test_category_encoding_inner(tmp_path, X_train, X_test, y_train, y_test, params, model_prefix):
+        # checks that category_encoders works for Dataset constructor
+        category_encoders_str = "target,count,target:0.5,raw"
+        train_data_1 = lgb.Dataset(X_train, label=y_train, category_encoders=category_encoders_str)
+        valid_data_1 = train_data_1.create_valid(X_test, label=y_test)
+
+        categorical_feature = [fidx for fidx in range(X_train.shape[1] // 2)]
+        expected_num_features = X_train.shape[1] + 3 * len(categorical_feature)
+
+        params.update({"categorical_feature": categorical_feature})
+        booster_1 = lgb.train(params, train_data_1, valid_sets=[valid_data_1], valid_names=["valid_data"])
+        np.testing.assert_equal(train_data_1.num_feature(), expected_num_features)
+        np.testing.assert_equal(valid_data_1.num_feature(), expected_num_features)
+        np.testing.assert_equal(len(train_data_1.get_feature_name()), expected_num_features)
+        np.testing.assert_equal(booster_1.num_feature(), expected_num_features)
+        pred_1 = booster_1.predict(X_test)
+        pred_contrib_1 = booster_1.predict(X_test, pred_contrib=True)
+        tmp_dataset = str(tmp_path / 'category_encoding_{}_temp_dataset.bin'.format(model_prefix))
+        train_data_1.save_binary(tmp_dataset)
+
+        # checks that Dataset with category_encoders can be saved to and load from file
```

Comment:
Seems this comment should be 3 lines upper (before `tmp_dataset = `).

Reply:
Resolved via a25f323